eventlet/greenio/py3.py

```diff
@@ -85,7 +85,7 @@ def read(self, size=-1):
             except OSError as e:
                 if get_errno(e) not in SOCKET_BLOCKING:
                     raise IOError(*e.args)
-            self._trampoline(self, read=True)
```

Comment:
this works either way, but i think having it inside the `except` clause is more semantically correct (and copying this is what caused the error below)